python/ray/serve/config.py

```diff
@@ -84,6 +91,33 @@ def replicas_settings_valid(cls, max_replicas, values):
 
         return max_replicas
 
+    @validator("policy", always=True)
+    def serialize_policy(cls, policy, values):
+        """Serialize policy to be cloudpickled and base64 encoded utf8 string.
+
+        If policy is a string, it is possible to be already serialized when constructed
+        from protobuf. In this case, we just return the already serialized string. Else
+        we import the policy and serialize it before return.
+        """
+        if isinstance(policy, Callable):
+            imported_policy = policy
+        elif isinstance(policy, str):
+            try:
+                assert b64encode(b64decode(policy)).decode("utf-8") == policy
+                return policy
+            except binascii.Error:
+                imported_policy = import_attr(policy)
+            except AssertionError:
+                imported_policy = import_attr(policy)
+        else:
+            return policy
+
+        return b64encode(cloudpickle.dumps(imported_policy)).decode("utf-8")
+
+    def get_policy(self) -> Callable:
```

Comment:
Could you add a unit test for this where the custom policy isn't provided inside the Ray package? We should make sure this works nicely with runtime environments.

Reply:
The runtime env was only working during my test because I was using the same name for the custom scaling function. Will go back to the drawing board for it.

Reply:
Talk with Shreyas and Ed offline. For now we will make a hard requirement for the custom autoscaling policy to exist in the cluster env. We will add a follow up when if the user complains about using this with runtime env. 

And in that case we can refactor the Autoscaling Manager into a Ray actor running in runtime env and constantly update an instance variable for the deployment replica num which controller can fetch and update.